internal/resource_remote_file.go

```diff
@@ -69,17 +72,73 @@ func (o remoteFileModel) AttrTypes() map[string]attr.Type {
 		"source_url":         types.StringType,
 		"source_remote_file": types.StringType,
 		"checksum":           types.StringType,
+		"extract":            types.ObjectType{}.WithAttributeTypes(remoteFileExtractModel{}.AttrTypes()),
 		"authentication":     types.ObjectType{}.WithAttributeTypes(remoteFileModelAuthenticationsModel{}.AttrTypes()),
 		"polling":            types.ObjectType{}.WithAttributeTypes(remoteFilePollingModel{}.AttrTypes()),
 	}
 }
 
+type remoteFileExtractModel struct {
+	Enable          types.Bool   `tfsdk:"enable"`
+	DestinationPath types.String `tfsdk:"destination_path"`
+	Password        types.String `tfsdk:"password"`
+	DeleteArchive   types.Bool   `tfsdk:"delete_archive"`
+	Overwrite       types.Bool   `tfsdk:"overwrite"`
+}
+
+func (o remoteFileExtractModel) ResourceAttributes() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"enable": schema.BoolAttribute{
+			Required:            true,
+			MarkdownDescription: "Enable extraction",
+		},
+		"destination_path": schema.StringAttribute{
```

Comment:
I am doubtful about this field.
When using the output of this resource, should we use `resource.destination_path` or `resource.extract.destination_path`, this would depends on the value of `resource.extract.enable` which make it hard to use.
I suggest to extract the file to the already existing `destination_path`, and maybe introduce a `extract.temporary_download_path` which we would be the path where to download the file.
wdyt?

Reply:
and set `delete_archive` to `true` by default?